mirgecom/fluid.py

```diff
@@ -427,9 +431,13 @@ def species_mass_fraction_gradient(discr, cv, grad_cv):
     """
     nspecies = len(cv.species_mass)
     y = cv.species_mass / cv.mass
-    return (1/cv.mass)*make_obj_array([grad_cv.species_mass[i]
+    obj_ary = (1/cv.mass)*make_obj_array([grad_cv.species_mass[i]
                                        - y[i]*grad_cv.mass
                                        for i in range(nspecies)])
+    grad_y = np.empty(shape=(nspecies, cv.dim), dtype=object)
+    for idx, v in enumerate(obj_ary):
+        grad_y[idx] = v
+    return grad_y
```

Comment:
Same question: why not just return the `obj_ary`?

Reply:
Just resolving so we can focus on the one convo.